PR - Solicitada por pedro para feedback

RodriguezMarioUTN opened the pull request:
PR - Solicitada por pedro para feedback

RodriguezMarioUTN committed:
Start with Unitest by Client.java and ClientServiceImpl.java using mockito

RodriguezMarioUTN committed:
add test of example by pedro today

RodriguezMarioUTN committed:
Solution Can't not resolve import run mokito and add utest

PedroAguiar commented on JUnitBasico/ShoppingCart/src/test/java/ClientTest.java:
```diff
@@ -1,9 +1,13 @@
 
 public class ClientTest {
-    private Client mario = new Client(99,"Mario","Rodriguez","dec");
+    List
```
? Esto no compila

PedroAguiar commented on shopping-cart-mario/src/test/java/com/shoppingcart/shoppingcartmario/model/ClientTest.java:
```diff
@@ -0,0 +1,32 @@
+
+public class ClientTest {
+
+    @InjectMocks
```
No hay fields con @Mock que vayan a ser injectados por el @InjectMocks

PedroAguiar commented on shopping-cart-mario/src/test/java/com/shoppingcart/shoppingcartmario/service/impl/ClientServiceImplTest.java:
```diff
@@ -0,0 +1,77 @@
+//    public void updateClientTest() {
+//        Client client = ModelUtils.createDefaultClient();
+//        when(mockClientReository.update(client.getFirstName(), client.getLastName(), client.getDescription(), client.getId())).thenReturn(ModelUtils.createDefaultClient());
+//        clientService.updateClient(ModelUtils.createDefaultClient());
```
Si le pasas un nuevo objeto al update y estas haciendo when(algo) sobre otro objeto, los asserts nunca van a pasar

PedroAguiar commented on shopping-cart-mario/src/test/java/com/shoppingcart/shoppingcartmario/util/ModelUtils.java:
```diff
@@ -0,0 +1,28 @@
+    public static List<Client> createListDefaultClient() {
+        List<Client> clients = new ArrayList<>();
+        clients.add(createDefaultClient());
+    return clients;
```
cuidado con la indentacion

RodriguezMarioUTN committed:
Correcciones Feedback de la PR listas

RodriguezMarioUTN committed:
Finished the Unit Test of Client

RodriguezMarioUTN committed:
Corrections completed TP 2

Corrections completed TP 2, now perfectly establishes the relationships and all the project FK

RodriguezMarioUTN committed:
Utest Ready!

Utest Ready!
